extension/src/bridge/host.ts

```diff
@@ -7,16 +7,33 @@ interface Request {
 
 export default class BridgeHost {
   private provider: Eip1193Provider
+  private connection: Connection
   private source: WindowProxy | undefined
 
-  constructor(provider: Eip1193Provider) {
+  constructor(provider: Eip1193Provider, connection: Connection) {
     this.provider = provider
+    this.connection = connection
   }
 
   setProvider(provider: Eip1193Provider) {
     this.provider = provider
   }
 
+  setConnection(connection: Connection) {
+    if (connection?.avatarAddress !== this.connection?.avatarAddress) {
```

Comment:
```suggestion
    if (connection.avatarAddress !== this.connection.avatarAddress || connection.chainId !== this.connection.chainId) {
```

Another connection might use the same avatar address on a different network

Reply:
The optional chaining operator seems unnecessary. Let put trust into our type definitions.

Reply:
> Another connection might use the same avatar address on a different network

Well spotted! Will fix that.